source/whats-new.txt

```diff
@@ -0,0 +1,128 @@
+.. _cpp-whats-new:
+
+==========
+What's New
+==========
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: update, new feature, deprecation, upgrade, driver
+
+Learn what's new in:
+
+* :ref:`Version 3.10 <version-3.10>`
+* :ref:`Version 3.9 <version-3.9>`
+
+.. _upcoming-breaking-changes:
+
+.. _version-3.10:
+
+What's New in 3.10
+------------------
+
+.. warning:: Breaking Changes
```

Comment:
```suggestion
.. warning:: Breaking Changes (Build System)
```

I would like to establish a pattern/precedent of distinguishing _API_ breaking changes from _build system/packaging_ breaking changes. Currently, all breaking changes listed in this PR are _build system_ breaking changes.

Although we do not yet have any API breaking changes in the current list of changes, I believe we should start applying this pattern now rather than later to be consistent with [wording in our other doc pages](https://github.com/mongodb/docs-cpp/pull/43#discussion_r1686933180).

An example of expected ordering and contents:

> [!WARNING]
> **Breaking Changes (API)**
>
> - Changes concerning the API of the C++ Driver libraries.
> - Addressing incompatibilities may require modifying C++ source code in user projects.
> - Removed function `func(A)`. Use `func(A, B)` instead.
> - Changed redeclaration of `bsoncxx::foo` from `v1::foo` to `v2::foo`. Qualify `bsoncxx::foo` as `bsoncxx::v1::foo` to continue using the old interface.

> [!WARNING]
> **Breaking Changes (ABI)**
> - Changes concerning the ABI of the C++ Driver libraries.
> - Addressing incompatibilites may require rebuilding user projects (modifying C++ source code may not be necessary).
> - Removed exported symbol for `v1::func(A)`. Use `v2::func(A)` instead.

> [!WARNING]
> **Breaking Changes (Build System)**
>
> - Changes concerning the build system or packaging of the C++ Driver libraries.
> - Addressing incompatibilities may require modifying build system configurations or package/dependency management settings in user projects (modifying or rebuilding C++ source code may not be necessary).
> - Removed configuration variable `ENABLE_FOO`.
> - Dropped support for platform X / compiler B / architecture C.

These three kinds of breaking changes have different stability requirements, conditions, and corresponding impact on users. I believe it is worth distinguishing them accordingly.